dlt/extract/incremental/__init__.py

```diff
@@ -578,6 +609,34 @@ def get_incremental_arg(sig: inspect.Signature) -> Optional[inspect.Parameter]:
                 break
         return incremental_param
 
+    @staticmethod
+    def inject_implicit_incremental_arg(
```

Comment:
Why we need an explicit method to inject Incremental passed from the decorator? we already have one that is used when `apply_hints` is used. (via `_set_hints` and later `bind` in the incremental). is there a problem with the precedence ie. to make sure that default value had it? IMO we can ignore such things and rely on the precedence from `apply_hints`

Reply:
Yeah might be getting into overly complicated workarounds here. I'll see about simplifying this.  
I got into issues where it's not possible to override the initial hint with an argument cause the hint always takes precedence. There were some cases as well where the hint does not get injected into the default incremental arg, it was lost when resource is cloned I think.